File: js/__tests__/db_link_list.test.js
const DBLinkList = require('../data_structures/db_link_list')

describe('双向链表测试', () => {

    let li;

    beforeEach(() => {
        li = new DBLinkList();
    })

    it('insert', () => {
        li.insert('hello');
        li.insert('world', 'hello');
        expect(li.toArr()).toEqual(['hello', 'world'])
    })

    it('remove', () => {
        li.insert('hello');
        li.insert('world'); // 默认头部插入暂时有点问题
        li.insert('bear', 'hello')
        expect(li.toArr()).toEqual(['world', 'hello', 'bear'])
        li.remove('hello');
        expect(li.head.next.item).toEqual('world')
        expect(li.toArr()).toEqual(['world', 'bear'])
    })

    it('reverse', () => {
        li.insert('hello');
        li.insert('world', 'hello')
        li.insert('bear', 'hello')
        expect(li.toArr()).toEqual(['hello', 'bear', 'world'])
        expect(li.toReverseArr()).toEqual(['world', 'bear', 'hello'])
    })

})

File: py/__tests__/db_linklist_test.py
import unittest
from data_structures.db_linklist import DBLinkList


class TestDbLinkList(unittest.TestCase):
    # 插入方法测试
    def test_insert(self):
        li = DBLinkList()
        li.insert('hello', 'head')
        li.insert('world', 'hello')
        li.insert('xiong', 'world')
        li.insert('bear')
        self.assertEqual(li.to_arr(), ['bear', 'hello', 'world', 'xiong'])

    # 删除方法
    def test_remove(self):
        li = DBLinkList()
        li.insert('hello', 'head')
        li.insert('bear', 'hello')
        li.insert('xiong', 'bear')
        self.assertEqual(li.to_arr(), ['hello', 'bear', 'xiong'])
        li.remove('bear')
        self.assertEqual(li.to_arr(), ['hello', 'xiong'])

    # 测试倒序方法
    def test_reverse(self):
        li = DBLinkList()
        li.insert('hello', 'head')
        li.insert('bear', 'hello')
        li.insert('xiong', 'hello')
        self.assertEqual(li.to_arr(), ['hello', 'xiong', 'bear'])
        self.assertEqual(li.to_reverse_arr(), ['bear', 'xiong', 'hello'])
